cmd/client/main.go

```diff
@@ -19,40 +20,9 @@ import (
 // The entry point of the program
 func main() {
 
-	// A start index greater than the end index indicates the program should be run once
-	// with no parameters.
-	var start int64 = 0
-	var end int64 = -1
-
-	var err error
-
-	// The command line arguments. args[0] is the name of the program.
-	args := os.Args
-
-	// If the right number of arguments weren't passed, ask for them and exit.
-	if len(args) < 3 {
-		fmt.Println("Please pass the address of the supervisor and a file to run, and an optional range of parameters.")
-		fmt.Println("Example: http://stu.cs.jmu.edu:4001 fun_code.py [1-10]")
-		os.Exit(1)
-	}
-
-	// Collect arguments if they're given
-	if len(args) == 4 {
-		split := strings.Split(args[3], "-")
-		if len(split) != 2 {
-			fmt.Println("Please give the parameter range as two dash-delimited numbers. For example, 1-100")
-			os.Exit(1)
-		}
-
-		start, err = strconv.ParseInt(split[0], 10, 64)
-		check(err)
-
-		end, err = strconv.ParseInt(split[1], 10, 64)
-		check(err)
-	}
-
+	hostName, fullFileName, start, end, args, runs := parseCommandLine()
```

Comment:
One more thing, could we consider passing these things in by reference to parseCommandLine? This many returns is kind of gross.

Reply:
Yeah I was thinking the same thing